ast/compile_test.go

```diff
@@ -110,6 +110,16 @@ func TestCompilerExample(t *testing.T) {
 	assertNotFailed(t, c)
 }
 
+func TestCompilerWithStageAfter(t *testing.T) {
+	c := NewCompiler().WithStageAfter("CheckRecursion", mockStageFunctionCall)
+	m := MustParseModule(testModule)
+	c.Compile(map[string]*Module{"testMod": m})
+
+	if !c.Failed() {
+		t.Errorf("Expected compilation error")
```

Comment:
When testing error paths, it's a good idea to check for the specific condition in the error instead of just checking _if some_ error occurred. We can merge this as is, but something to keep in mind in future.